From c55870ceaf675ad2fba9605052ecf15237e2e4c4 Mon Sep 17 00:00:00 2001
From: Steve Dickson <SteveD@redhat.com>
Date: Thu, 26 Jun 2014 15:54:17 -0400
Subject: [net] sunrpc/xprtrdma: mount reports "Invalid mount option" if memreg mode not supported

Message-id: <1403798077-28799-59-git-send-email-steved@redhat.com>
Patchwork-id: 84923
O-Subject: [RHEL7.1 PATCH 58/78] xprtrdma: mount reports "Invalid mount option" if memreg mode not supported
Bugzilla: 1113248
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>

From: Chuck Lever <chuck.lever@oracle.com>

If the selected memory registration mode is not supported by the
underlying provider/HCA, the NFS mount command reports that there was
an invalid mount option, and fails. This is misleading.

Reporting a problem allocating memory is a lot closer to the truth.

Signed-off-by: Chuck Lever <chuck.lever@oracle.com>
Signed-off-by: Anna Schumaker <Anna.Schumaker@Netapp.com>
Upstream Id: cdd9ade711599e7672a635add0406080856f8b92
Bugzilla: 1113248
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/sunrpc/xprtrdma/verbs.c b/net/sunrpc/xprtrdma/verbs.c
index a352798..7c7e9b4 100644
--- a/net/sunrpc/xprtrdma/verbs.c
+++ b/net/sunrpc/xprtrdma/verbs.c
@@ -513,7 +513,7 @@ rpcrdma_ia_open(struct rpcrdma_xprt *xprt, struct sockaddr *addr, int memreg)
 #if RPCRDMA_PERSISTENT_REGISTRATION
    memreg = RPCRDMA_ALLPHYSICAL;
 #else
-   rc = -EINVAL;
+   rc = -ENOMEM;
    goto out2;
 #endif
   }
@@ -554,9 +554,9 @@ rpcrdma_ia_open(struct rpcrdma_xprt *xprt, struct sockaddr *addr, int memreg)
   }
   break;
  default:
-  printk(KERN_ERR "%s: invalid memory registration mode %d\n",
-    __func__, memreg);
-  rc = -EINVAL;
+  printk(KERN_ERR "RPC: Unsupported memory "
+    "registration mode: %d\n", memreg);
+  rc = -ENOMEM;
   goto out2;
  }
  dprintk("RPC:       %s: memory registration strategy is %d\n",
-- 
1.7.1